config.go

```diff
@@ -46,6 +47,12 @@ func decodeConfig(r io.Reader, c *config) error {
 // Hence, the priority order is the reverse of the search order - i.e., the
 // CWD has the highest priority.
 func (c *config) Discover() error {
+	// If we are already inside a plugin process we should not need to
+	// discover anything.
+	if os.Getenv(plugin.MagicCookieKey) != "" {
```

Comment:
should we check that it equals `$PACKER_PLUGIN_MAGIC_COOKIE`? like https://github.com/mitchellh/packer/blob/master/packer/plugin/server.go#L42